tests/Middleware/LimitHandlersMiddlewareTest.php

```diff
@@ -0,0 +1,424 @@
+    {
+        $stream = new ThroughStream();
+        $request = new ServerRequest(
+            'POST',
+            'http://example.com/',
+            array(),
+            new HttpBodyStream($stream, 5)
+        );
+
+        $req = null;
+        $middleware = new LimitHandlersMiddleware(1);
+        $middleware($request, function (ServerRequestInterface $request) use (&$req) {
+            $req = $request;
+        });
+
+        $this->assertNotSame($request, $req);
+        $this->assertInstanceOf('Psr\Http\Message\ServerRequestInterface', $req);
+
+        $body = $req->getBody();
+        $this->assertInstanceOf('React\Stream\ReadableStreamInterface', $body);
+        /* @var $body \React\Stream\ReadableStreamInterface */
+
+        $this->assertEquals(5, $body->getSize());
+
+        $body->on('data', $this->expectCallableOnce('hello'));
+        $stream->write('hello');
+    }
+
+    public function testReceivesRequestsSequentially()
+    {
+        $request = new ServerRequest(
+            'POST',
+            'http://example.com/',
+            array(),
+            'hello'
+        );
+
+        $middleware = new LimitHandlersMiddleware(1);
+        $middleware($request, $this->expectCallableOnceWith($request));
+        $middleware($request, $this->expectCallableOnceWith($request));
+        $middleware($request, $this->expectCallableOnceWith($request));
+    }
+
+    public function testDoesNotReceiveNextRequestIfHandlerIsPending()
+    {
+        $request = new ServerRequest(
+            'POST',
+            'http://example.com/',
+            array(),
+            'hello'
+        );
+
+        $middleware = new LimitHandlersMiddleware(1);
+        $middleware($request, function () {
+            return new Promise(function () {
+                // NO-OP: pending promise
+            });
+        });
+
+        $middleware($request, $this->expectCallableNever());
+    }
+
+    public function testReceivesNextRequestAfterPreviousHandlerIsSettled()
+    {
+        $request = new ServerRequest(
+            'POST',
+            'http://example.com/',
+            array(),
+            'hello'
+        );
+
+        $deferred = new Deferred();
+        $middleware = new LimitHandlersMiddleware(1);
+        $middleware($request, function () use ($deferred) {
+            return $deferred->promise();
+        });
+
+        $deferred->reject(new \RuntimeException());
+
+        $middleware($request, $this->expectCallableOnceWith($request));
+    }
+
+    public function testPendingRequestCanBeCancelledAndForwardsCancellationToInnerPromise()
+    {
+        $request = new ServerRequest(
+            'POST',
+            'http://example.com/',
+            array(),
+            'hello'
+        );
+
+        $once = $this->expectCallableOnce();
+        $deferred = new Deferred(function () use ($once) {
+            $once();
+            throw new RuntimeException('Cancelled');
```

Comment:
Travis is crashing on this: https://travis-ci.org/reactphp/http/jobs/313374574#L512

Reply:
Duh, updated invalid reference just now :shipit: